superset-frontend/src/dashboard/components/DeleteComponentButton.tsx

```diff
@@ -16,25 +16,21 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { MouseEventHandler } from 'react';
 import Icons from 'src/components/Icons';
 import IconButton from './IconButton';
 
-const propTypes = {
-  onDelete: PropTypes.func.isRequired,
+type DeleteComponentButtonProps = {
+  onDelete: MouseEventHandler<HTMLDivElement>;
 };
 
-const defaultProps = {};
-
-export default class DeleteComponentButton extends React.PureComponent {
+class DeleteComponentButton extends React.PureComponent<DeleteComponentButtonProps> {
   render() {
-    const { onDelete } = this.props;
+    const { onDelete = () => {} } = this.props;
```

Comment:
I don;'t think you need to change it here.

Reply:
Alright, I've removed the default value.